packages/marshal/src/types.js

```diff
@@ -2,70 +2,317 @@
+ *      lisp/scheme gensyms. From JS it currently includes only
+ *      `Symbol.asyncIterator` (a so-called well-known symbol). From JS
+ *      it can include at most the well known symbols and the registered
+ *      symbols. (Because symbol registration names can collide with
+ *      well known symbol names, we will need another hilbert hotel
+ *      encoding to allow well known symbols and all registered symbols.)
+ *      From lisp, it can include symbols made from interned strings.
+ */
 
 /**
- * @typedef {*} Passable
+ * @typedef {*} PrimaryRemotable
+ * An object marked with `Far` consisting of methods that can be remotely
+ * invoked. A PrimaryRemotable has a fresh unforgeable identity, i.e.,
+ * an unforgeable identity created when the object was created.
  *
- * A Passable value that may be marshalled. It is classified as one of
- * PassStyle. A Passable must be hardened.
+ * A PrimaryRemotable, being a kind of Remotable, is pass-by-remote. When a
+ * PrimaryRemotable T is passed as an argument in a message to an object in
+ * another vat, it arrives as a Remote<T>.
+ */
+
+/**
+ * @template {PrimaryRemotable} T
+ * @typedef {*} Remote
+ * A `Remote<T>` is a Remotable object in one vat that represents a
+ * PrimaryRemotable of type T in another vat. The Remote has an unforgeable
+ * identity that is one-to-one with the unforgeable identity of its primary.
+ * If there is only one comm system between the two vats, then the identity of
+ * the Remote is not observably different than the identity of its primary.
  *
- * A Passable has a pass-by-copy superstructure. This includes the atomic
- * pass-by-copy primitives ("bigint" | "boolean" | "null" | "number" |
- * "string" | "undefined") and the composite pass-by-copy objects ("copyArray" |
- * "copyRecord" | "copyError"). The composite pass-by-copy objects that may
- * contain other Passables.
+ * A Remote, being a kind of Remotable, is pass-by-remote. When a Remote<T>
+ * is passed as an argument in a message to an object in another vat, it
+ * arrives as either a T or a Remote<T>, depending on whether the destination
+ * is the home vat of its primary.
+ */
+
+/**
+ * @template {PrimaryRemotable} T
+ * @typedef {T | Remote<T>} Remotable
+ * Corresponding PassStyle is "remotable"
  *
- * A Passable's pass-by-copy superstructure ends in PassableCap leaves. The
- * Passable can be further classified by the nature of these leaves. Since a
- * Passable is hardened, its structure and classification is stable --- its
- * structure and classification cannot change even if some of the objects are
- * proxies.
+ * A Remotable<T> is either the PrimaryRemotable T in its home vat, or is one
+ * of its Remote<T> representatives in other vats. The primary together with
+ * all of its remotes are, in aggregate, the Unum<T>, the abstact distributed
+ * object consisting of all these individual remotables. If there is only one
```

Comment:
Please do not consume the word `Unum` in this way.  This is not an unum (or, rather, it _is_ an unum, but it's a specific kind of unum out of a larger space of possibilities).

Reply:
ok. But I'm glad you agree it is an Unum. I expect it will be the base from which we grow richer Unum behaviors, and that these richer Unums will still fit as the collection of `Remotables` with the same cross-vat unforgeable identity. But I don't currently need a short word for that collection as a thing.

Reply:
So some time when we're less pressed for time, you and I should have a longer conversation about The Unum Nature.  I've built a lot of things using the unum pattern and almost none of them match the structure you are elaborating here.

Reply:
> So some time when we're less pressed for time, you and I should have a longer conversation about The Unum Nature. I've built a lot of things using the unum pattern and almost none of them match the structure you are elaborating here.

Please!